File: ws/src/main/scala/de/thm/ii/fbs/model/GlobalRole.scala
package de.thm.ii.fbs.model

/**
  * The global roles of users
  */
object GlobalRole extends Enumeration {
  /**
    * Admin manages everything.
    */
  val ADMIN: GlobalRole.Value = Value(0)
  /**
    * Moderator manages courses, docents and tutors.
    */
  val MODERATOR: GlobalRole.Value = Value(1)
  /**
    * Usual user.
    */
  val USER: GlobalRole.Value = Value(2)

  /**
    * Parse an int to a global role.
    * @param roleId The role id
    * @return Global role, where the id is mapped to its assigned enum type, if no enum type
    *         is assigned to the roleId provided, then the USER type is returned.
    */
  def parse(roleId: Int): GlobalRole.Value = roleId match {
    case 0 => GlobalRole.ADMIN
    case 1 => GlobalRole.MODERATOR
    case _ => GlobalRole.USER
  }
}

File: modules/fbs-identity-service/api/src/main/kotlin/de/thm/ii/fbs/fbs_identity_service/model/GlobalRole.kt
package de.thm.ii.fbs.fbs_identity_service.model

enum class GlobalRole(val id: Int) {
    ADMIN(0),
    MODERATOR(1),
    USER(2);

    companion object {
        fun parse(roleId: Int): GlobalRole {
            return when (roleId) {
                0 -> ADMIN
                1 -> MODERATOR
                else -> USER
            }
        }

        fun parse(roleName: String): GlobalRole {
            return when (roleName) {
                "ADMIN" -> ADMIN
                "MODERATOR" -> MODERATOR
                else -> USER
            }
        }
    }
}